src/Microsoft.DotNet.Tools.Compiler.Native/NativeCompileSettings.cs

```diff
@@ -1,53 +1,195 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
-
-using Microsoft.Dnx.Runtime.Common.CommandLine;
 using Microsoft.DotNet.Cli.Utils;
-using Microsoft.DotNet.Tools.Common;
 
 namespace Microsoft.DotNet.Tools.Compiler.Native
 {
     public class NativeCompileSettings
     {
-        public string LogPath { get; set; }
-        public string InputManagedAssemblyPath { get; set; }
-        
-        public string OutputDirectory { get; set; }
-        public string IntermediateDirectory { get; set; }
-        
+        private const BuildConfiguration DefaultBuiltType = BuildConfiguration.debug;
+        private const NativeIntermediateMode DefaultNativeModel = NativeIntermediateMode.ryujit;
+        private const ArchitectureMode DefaultArchitectureMode = ArchitectureMode.x64;
+
+        private string _inputManagedAssemblyPath;
+        private string _appDepSdkPath;
+        private string _ilcPath;
+        private string _outputDirectory;
+        private string _intermediateDirectory;
+        private string _logPath;
+        private string _ilcArgs;
+        private readonly List<string> _referencePaths;
+        private readonly List<string> _linkLibPaths;
+
+        public string LogPath
+        {
+            get { return _logPath; }
+            set { _logPath = Path.GetFullPath(value); }
+        }
+
+        public string InputManagedAssemblyPath
+        {
+            get
+            {
+                return _inputManagedAssemblyPath;
+            }
+            set
+            {
+                if(!File.Exists(value))
+                {
+                    throw new Exception($"Could not find the input managed assembly: {value}");
+                }
+
+                _inputManagedAssemblyPath = Path.GetFullPath(value);
```

Comment:
This would only be expected to be set once per run. Even if this is expensive, it seems reasonable to have this here.